source/extensions/http/cache/file_system_http_cache/cache_file_fixed_block.cc

```diff
@@ -0,0 +1,94 @@
+#include "source/extensions/http/cache/file_system_http_cache/cache_file_fixed_block.h"
+
+#include "source/common/common/assert.h"
+#include "source/common/common/safe_memcpy.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace FileSystemHttpCache {
+
+namespace {
+// The expected first four bytes of the header - if fileId() doesn't match ExpectedFileId
+// then the file is not a cache file and should be removed from the cache.
+// Beginning of file should be "CACH".
+std::array<char, 4> ExpectedFileId = {'C', 'A', 'C', 'H'};
```

Comment:
these need to be constexpr to avoid an init-order fiasco per style guide. But they don't need to be std::array; can't they be
```
constexpr absl::string_view ExpectedFileId("CACH"}; 
constexpr absl::string_view ExpectedCacheVersionId{"\0\0\0\0", 4|;

Reply:
std::array makes them explicitly four bytes long, known by the compiler. string_view makes them runtime four bytes long.